include/zen/scene/screen-layout.h

```diff
@@ -8,6 +8,10 @@ struct zn_screen;
 
 struct zn_screen_layout {
   struct wl_list screens;  // zn_screen::link
+
+  // this is emited when zn_screen added to zn_screen_layout.
+  struct wl_signal add_screen;  // (struct zn_screen *)
```

Comment:
```suggestion
  struct {
    new_screen; // (struct zn_screen *)
  } events;
```

Following the convention of wlroots, I feel that signals should be placed in events struct.
(I wanna fix `zn_display_system::switch_signal`.)